### PR TITLE
Remove "min size" constrains to available size in layouter

### PR DESCRIPTION
**Pull request by carldebilly**
This is to align with the UWP/WinUI behavior.
The Width/MinWidth restrictions are applied by the control itself,
not the layouting process.

GitHub Issue (If applicable): #

# Bugfix

## What is the current behavior?
The `Layouter` (and the equivalent for Wasm) was applying the `minSize` (calculated from `Width`, `MinWidth`, `Height`, `MinHeight`) on the `availableSize` during the `Arrange()` phase.

This was allowing the `availableSize` to grow when smaller than `minSize`.

## What is the new behavior?
The `availableSize` passed to the `ArrangeOverride()` of a control won't apply the `minSize` anymore.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [X] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [X] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [X] Associated with an issue (GitHub or internal)

Internal Issue (If applicable):
* https://nventive.visualstudio.com/Umbrella/_workitems/edit/168813 (this is the reference for the sample)
* https://nventive.visualstudio.com/Umbrella/_workitems/edit/168844 (tentatively)
* https://nventive.visualstudio.com/Umbrella/_workitems/edit/168858 (tentatively)
* https://nventive.visualstudio.com/Umbrella/_workitems/edit/168973 (tentatively)

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs [hunk 1]
```diff
@@ -98,8 +98,7 @@ public Size Measure(Size availableSize)
 					.NumberOrDefault(MaxSize)
 					.Subtract(marginSize)
 					.AtLeast(default) // 0.0,0.0
-					.AtMost(maxSize)
-					.AtLeast(minSize);
+					.AtMost(maxSize);
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs]: You should add a RuntimeTest which verifies this behavior

**Comment by jeromelaban** [comment]
@mergifyio backport release/beta/1.0

**Comment by jeromelaban** [comment]
@Mergifyio backport release/beta/2.0

**Comment by jeromelaban** [comment]
@mergify backport release/beta/2.0

**Pull request merged**